lib/flatpak-installation.c

```diff
@@ -1112,170 +1112,254 @@ get_metadata_progress (guint metadata_fetched,
+  gchar *formatted_bytes_transferred =
+    g_format_size_full (bytes_transferred, 0);
+  gchar *formatted_bytes_sec = NULL;
+
+  if (!bytes_sec) // Ignore first second
+    formatted_bytes_sec = g_strdup ("-");
+  else
+    formatted_bytes_sec = g_format_size (bytes_sec);
+
+  *out_transferred = formatted_bytes_transferred;
+  *out_bytes_per_sec = formatted_bytes_sec;
 }
 
 static void
 progress_cb (OstreeAsyncProgress *progress, gpointer user_data)
 {
   FlatpakProgressCallback progress_cb = g_object_get_data (G_OBJECT (progress), "callback");
   guint last_progress = GPOINTER_TO_UINT (g_object_get_data (G_OBJECT (progress), "last_progress"));
+  g_autofree gchar *formatted_bytes_sec;
+  g_autofree gchar *formatted_bytes_transferred;
   GString *buf;
   g_autofree char *status = NULL;
   guint outstanding_fetches;
+  guint outstanding_delta_fetches;
   guint outstanding_metadata_fetches;
   guint outstanding_writes;
-  guint n_scanned_metadata;
-  guint fetched_delta_parts;
-  guint total_delta_parts;
   guint64 bytes_transferred;
+  guint64 fetched_delta_part_size;
   guint64 total_delta_part_size;
   guint outstanding_extra_data;
   guint64 total_extra_data_bytes;
   guint64 transferred_extra_data_bytes;
-  guint fetched;
+  guint64 total;
+  guint64 total_transferred;
   guint metadata_fetched;
-  guint requested;
   guint64 current_time;
+  guint64 fetched_content_bytes;
+  guint64 total_content_bytes;
   guint new_progress = 0;
   gboolean estimating = FALSE;
   gboolean downloading_extra_data;
 
+  buf = g_string_new ("");
+
   /* The heuristic here goes as follows:
-   *  - If we have delta files, grow up to 45%
-   *  - Else:
-   *     - While fetching metadata, grow up to 10%
-   *     - Then, while fetch the file objects, grow up to:
-   *        - 45% if we have extra data to download
-   *        - 90% otherwise
-   * - Writing the objects goes from 52% to 55%, or 97% to 100%
-   * - Extra data, if present, will go from 55% to 100%
+   *  - While fetching metadata, grow up to 5%
+   *  - Download goes up to 97%
+   *  - Writing objects adds the last 3%
+   *
+   *
+   * Meaning of each variable:
+   *
+   *   Status:
+   *    - status: only sent by OSTree when the pull ends (with or without an error)
+   *
+   *   Fetches:
+   *    - fetched: sum of content + metadata fetches
+   *    - requested: sum of requested content and metadata fetches
+   *    - bytes_transferred: every and all tranferred data (in bytes)
+   *    - metadata_fetched: the number of fetched metadata objects
+   *    - outstanding_fetches: missing fetches (metadata + content + deltas)
+   *    - outstanding_delta_fetches: missing delta-only fetches
+   *    - outstanding_metadata_fetches: missing metadata-only fetches
+   *    - fetched_content_bytes: the estimated downloaded size of content (in bytes)
+   *    - total_content_bytes: the estimated total size of content, based on average bytes per object (in bytes)
```

Comment:
This is odd. I know that it's confusing to juggle all these vars around but I usually prefer to not have comments like this as it is very easy to get them deprecated (someone will eventually touch the code without updating this list). A much better alternative is to give comprehensive names to vars, e.g. _metadata_fetched_ -> _total_metadata_objs_fetched_ , etc.